gossip/minimal/goshawk.go

```diff
@@ -304,6 +448,21 @@ func (hawk *Goshawk) Fly(ctx context.Context) {
 		}(origin)
 	}
 
+	// Flush scan state occasionally in case of abrupt termination.
+	ticker := time.NewTicker(hawk.fetchOpts.FlushInterval)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				if err := hawk.fetchOpts.State.Flush(ctx); err != nil {
+					glog.Errorf("State flush failed: %v", err)
```

Comment:
I assume this isn't an actual problem. If it restarts it will just do more work going over entries it's seen before? If not then it should maybe exit.

Reply:
Yeah, it just means the next run will repeat stuff. Added a comment.